### DISCUSSION TITLE
Wiki.js inaccessible after upgrade (2.5.260 > 2.5.268)

### DISCUSSION THREAD

**Discussion by Congenial-GmbH** [requarks/wiki#4837] [Error / Bug Report] [answered]
Hi,

Can anyone help me with this?

General configuration:
----------------------- 
OS:
	Ubuntu 20.04.3 LTS and Docker
Wiki.js version:
	2.5.268
Database engine:
	PostgreSQL 9.6`

Issue:
------
After the upgrade from wiki.js version 2.5.60 to 2.5.268 my wiki is inaccessible:
1. HTTPS-Access (https://mywiki.net/)
	Browser Message: "Unable to connect, An error occured during a connection to mywiki.net"
2. HTTP-Access (http://mwiki.net or http://www.wiki.net)
	Browser Message: "Wiki.js - Unknown authentication strategy "jwt"

Additional Information:
-----------------------
1. Ports 22, 80, 443 are open and accessible from anywhere
2. TLS configuration (environment variables)
	SSL_ACTIVE=1
      LETSENCRYPT_DOMAIN and LETSENCRYPT_EMAIL are still defined correctly defined.

Hypothesis:
-----------
- Authentication to my wiki is only allowed over https (SSL_ACTIVE=1). This is why authentication fails over http.
- HTTPS-Access might fail, because of an expired or corrupt certificate.
  (Environment variables LETSENCRYPT_DOMAIN, LETSENCRYPT_EMAIL are still corrrectly defined.)

Resolution attempts:
--------------------
1. https://docs.requarks.io/troubleshooting
   (Error: Unknown authentication strategy "jwt")
   Result: Reloading site does not resolve the strategy
2. https://docs.requarks.io/troubleshooting
   (How to manually disable HTTPS / SSL Redirection)
	Docker command
		docker exec db psql -U wiki -d wiki -c "DELETE FROM settings WHERE key = 'server';"
	returns
		DELETE 0
	Issue persists (after restarting wiki by docker restart wiki)
3. Tried to set environment variable SSL_ACTIVE=0 in docker wiki docker container
	Resutl: I did not manage to achieve this

**Comment by Congenial-GmbH** [comment]
I'd also be happy just with a hint that I can try out to regain access.

**Comment by Congenial-GmbH** [comment]
Update:
I managed to set the environment variable SSL_ACTIVE=0  based on the instructions on https://docs.requarks.io/install/ubuntu:
___
$ docker stop wiki
$  docker rm wiki
$ docker create --name=wiki -e SSL_ACTIVE=0 -e DB_TYPE=postgres -e DB_HOST=db -e DB_PORT=5432 -e DB_PASS_FILE=/etc/wiki/.db-secret -v /etc/wiki/.db-secret:/etc/wiki/.db-secret:ro -e DB_USER=wiki -e DB_NAME=wiki -e UPGRADE_COMPANION=1 --restart=unless-stopped -h wiki --network=wikinet -p 80:3000 -p 443:3443 requarks/wiki:2
___

After this, I could log into my wiki via an insecure connection.

However, if I configure "Automatic HTTPS with Let's Encrypt" (https://docs.requarks.io/install/ubuntu) again, the log (docker logs wiki) shows the following error messages and I cannot connect to my wiki (via https):

___
(node:1) [DEP0148] **DeprecationWarning: Use of deprecated folder mapping "./public/" in the "exports" field module resolution of the package at /wiki/node_modules/extract-files/package.json.**
Update this package.json to use a subpath pattern like "./public/*".
(Use `node --trace-deprecation ...` to show where the warning was created)
2022-01-04T13:37:04.196Z [MASTER] info: HTTP Server: [ RUNNING ]
2022-01-04T13:37:04.476Z [MASTER] info: Telemetry is active: [ OK ]
2022-01-04T13:37:04.641Z [MASTER] info: (LETSENCRYPT) Generating certificate signing request (CSR)...
2022-01-04T13:37:04.659Z [MASTER] info: (LETSENCRYPT) CSR generated successfully [ OK ]
2022-01-04T13:37:04.659Z [MASTER] info: (LETSENCRYPT) Requesting certificate from Let's Encrypt...
2022-01-04T13:37:04.670Z [MASTER] info: (LETSENCRYPT) Setting HTTP challenge for consultability.net: [ READY ]
2022-01-04T13:37:04.671Z [MASTER] info: (LETSENCRYPT) Waiting for challenge to complete...
2022-01-04T13:37:09.682Z [MASTER] info: (LETSENCRYPT) Removing HTTP challenge: [ OK ]
2022-01-04T13:37:09.683Z [MASTER] info: (LETSENCRYPT) Removing HTTP challenge: [ OK ]
2022-01-04T13:37:09.684Z [MASTER] warn: (LETSENCRYPT) Error: connect ECONNREFUSED 127.0.0.1:80
2022-01-04T13:37:09.684Z [MASTER] **error: connect ECONNREFUSED 127.0.0.1:80**
___

Can anyone give me a hint, how I can fix this?

**Comment by STEAMWriter** [comment]
I had a similar problem accessing Wikidocs around the time as @Congenial-GmbH . I kept receiving a 502 error, but had changed nothing. The error disappeared. Access is restored. Now, edits are saved in the background of an existing page, but don't display on the viewer facing page. 

When I create a new page, the page and any edited text is visible. No, the permissions have not changed.

**Comment by Congenial-GmbH** [comment, marked as answer]
Thanks for your Feedback @STEAMWriter .

I finally resolved the issue by setting up a new cloud server and migrating the data from the old wiki to the new one following the transfer-manual on
https://docs.requarks.io/install/transfer

... and then setting up TLS/SSL Encryption according to Section *Automatic HTTPS with Let's Encrypt (optional)* on
https://docs.requarks.io/en/install/ubuntu

Even though the problem is solved, I would have liked to understand the reason for the network problem.

**Reply by STEAMWriter** [reply]
Thanks @Congenial-GmbH. 

I will pass the information to the developers. Maybe they can follow your directions or use some part of your directions to fix the editing issue.
